File: rss-reader/app/src/main/java/com/abubusoft/kripton/examples/rssreader/service/persistence/FilterType.java
package com.abubusoft.kripton.examples.rssreader.service.persistence;

public enum FilterType {
    ALL(""),
    UNREAD("read=0"),
    READ("read=1");

    FilterType(String sql) {
        this.sql=sql;
    }

    String sql;

    public String getSql() {
        return sql;
    }
}

File: kotlin-rss-reader/app/src/main/java/com/abubusoft/kripton/example/rssreader/service/persistence/FilterType.kt
package com.abubusoft.kripton.example.rssreader.service.persistence

enum class FilterType private constructor(sql: String) {
    ALL(""),
    UNREAD("read=0"),
    READ("read=1");

    var sql: String
        internal set

    init {
        this.sql = sql
    }
}